### PR TITLE
Add support for UUID columns to postgres & sqlite migrations.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

**Comment by nuttycom** [comment]
@tathougies this is my first attempt at support for a new data type, so it's likely I'm missing some necessary instances or something. Please let me know what else might need to be added to make this workable.

File: beam-postgres/Database/Beam/Postgres/Syntax.hs [hunk 1]
```diff
@@ -417,6 +417,7 @@ instance IsSql92DataTypeSyntax PgDataTypeSyntax where
   dateType = PgDataTypeSyntax (PgDataTypeDescrOid (Pg.typoid Pg.date) Nothing) (emit "DATE")
   timeType prec withTz = PgDataTypeSyntax (PgDataTypeDescrOid (Pg.typoid Pg.time) Nothing) (emit "TIME" <> pgOptPrec prec <> if withTz then emit " WITH TIME ZONE" else mempty)
   timestampType prec withTz = PgDataTypeSyntax (PgDataTypeDescrOid (Pg.typoid Pg.timestamp) Nothing) (emit "TIMESTAMP" <> pgOptPrec prec <> if withTz then emit " WITH TIME ZONE" else mempty)
+  uuidType = PgDataTypeSyntax(PgDataTypeDescrOid (Pg.typoid Pg.uuid) Nothing) (emit "uuid")
```
Comment (nuttycom) [inline beam-postgres/Database/Beam/Postgres/Syntax.hs]: The one thing I'm not certain about with this is that it requires adding the 'uuid-ossp' extension to make the "uuid" type available. This extension has been part of the standard postgres installation for a number of years, but you have to run `CREATE EXTENSION` for it explicitly for the schema where it is needed. Is there any mechanism whereby one could ensure that the extension has been enabled before attempting to create a UUID column?

File: beam-core/Database/Beam/Backend/SQL/SQL92.hs [hunk 2]
```diff
@@ -177,6 +177,7 @@ class IsSql92DataTypeSyntax dataType where
   dateType :: dataType
   timeType :: Maybe Word -> Bool {-^ With time zone -} -> dataType
   timestampType :: Maybe Word -> Bool {-^ With time zone -} -> dataType
+  uuidType :: dataType
```
[the remark this answers is not in the document]
Comment (tathougies) [reply beam-core/Database/Beam/Backend/SQL/SQL92.hs]: One more thing.... also add a `uuid :: Db.DataType PgDataTypeSyntax UUID` declaration in `Database.Beam.Postgres.Migrate`

You can see examples of other postgres-specific data types there.

**Comment by tathougies** [comment]
@nuttycom  Looks good overall 👍

File: beam-postgres/Database/Beam/Postgres/Syntax.hs [hunk 3]
```diff
@@ -1150,4 +1151,4 @@ instance HasDefaultSqlDataTypeConstraints PgColumnSchemaSyntax (SqlSerial Int)
 
 instance HasDefaultSqlDataType PgDataTypeSyntax Int64 where
```
[the remark this answers is not in the document]
Comment (nuttycom) [reply beam-postgres/Database/Beam/Postgres/Syntax.hs]: Okay. I've updated the PR; there was already a HasSqlValueSyntax instance for UUID.

**Comment by tathougies** [comment]
LGTM!

**Pull request merged**